### PR TITLE
Run pi x

### LINKED ISSUE #4: run_pi_x
add all run_pi_x in the root folder, like for the balance case

Do you think you can handle it?

### PR DESCRIPTION
**Pull request by aremazeilles**
Fix #4 

Could you pull that branch `run_pi_x` locally and have a look at it?

After making some changes in the code, I could launch the borg scale:

```shell
../run_pi_borgScale10 borgScale10/tests/data/input/questionnaire_borgScale10.csv outdir
``` 

Now the pi score that I get:

```yaml
type: 'vector'
value: [96.00000, 72.00000, 88.00000, 40.00000, 144.00000, 160.00000, 108.00000,
 105.00000, 107.00000, 101.00000, 114.00000, 116.00000, 108.00000, 105.00000, 10
7.00000, 101.00000, 114.00000, 116.00000, 108.00000, 105.00000, 107.00000, 101.0
0000, 114.00000, 116.00000, 108.00000, 105.00000, 107.00000, 101.00000, 114.0000
0, 116.00000, 108.00000, 105.00000, 107.00000, 101.00000, 114.00000, 116.00000, 
108.00000, 105.00000, 107.00000, 101.00000, 114.00000, 116.00000, 108.00000, 105
.00000, 107.00000, 101.00000, 114.00000, 116.00000, 108.00000, 105.00000, 107.00
000, 101.00000, 114.00000, 116.00000]
```

is different to the one stored in `borgScale10/tests/data/output/Local_Score.yaml`: 

```yaml
type: 'vector'
value: [12.00000, 0.00000, 1.00000, 1.00000, 1.00000, 2.00000, 1.00000, 5.00000,
 1.00000]
```
can you check this?

If possible, could you commit and push related change to this concrete branch, and inform me onto this pull request page?

Note that this is a draft pull request: a sort of discussion flow around a branch that is a work in progress.

### CODE REVIEW

**Comment by aremazeilles** [comment]
With the following changes I adjusted a bit the code to make the run_pi code working.
With the range of motion, none of the provided files enabled me to get the expected result stored in the output folder.  Could you check this, eventually update the output file, and possibly remove the other input files that are not related to the kept output, and update the general Readme with the appropriate command? thanks

**Comment by jamatics** [comment]
I will take a look at these issues on next Tuesday.
Thanks,
Jawad

**Review by jamatics — COMMENTED** [review]
OK thanks for correcting

File: README.md [hunk 1]
```diff
@@ -70,6 +70,12 @@ with:
 
 ### Borg scale
 
+Assuming folder `outdir` exists:
+
+```shell
+../run_pi_borgScale10 borgScale10/tests/data/input/questionnaire_borgScale10.csv outdir
```
Comment (jamatics) [inline README.md]: "../run_pi_borgScale10 borgScale10/tests/data/input/subject_01_questionnaire_borgScale10.csv outdir"
There was a bug in selecting the questionnaire input file. The input file is "subject_01_questionnaire_borgScale10.csv"

File: chrono/src/computePI.m [hunk 2]
```diff
@@ -76,7 +73,7 @@
 
 [filepath, name, ext] = fileparts(csv_file);
 
-filename = strcat(result_dir, "/", "Local_Score", ".yaml");
+filename = strcat(result_dir, "/", "pi_chrono", ".yaml");
```
Comment (jamatics) [inline chrono/src/computePI.m:76]: Thank for this. Good.

**Review by jamatics — COMMENTED** [review]
> With the following changes I adjusted a bit the code to make the run_pi code working.
> With the range of motion, none of the provided files enabled me to get the expected result stored in the output folder. Could you check this, eventually update the output file, and possibly remove the other input files that are not related to the kept output, and update the general Readme with the appropriate command? thanks

Hi Anthony, 

I find a small bug that the available input file that can generate the desired result "subject_03_trail_jointAngles.csv". I have changed name of the file in the input folder (rangeofMotion/tests/data/input/subject_03_trail_01_jointAngles.csv) to subject_03_run_01_jointAngles.csv. I have also removed the unnecessary files from input folder. Please confirm,  if it can resolve the issue. 
Thanks. 
Jawad

**Comment by aremazeilles** [comment]
Still having a difference with the range of motion:

```shell
./run_pi_rangeofMotion rangeofMotion/tests/data/input/subject_03_run_01_jointAngles.csv outdir
diff outdir/pi_rangeofmotion.yaml rangeofMotion/tests/data/output/Local_Score.yaml 
2c2
< value: [2.00000, -36.61945, -62.87399, 2.00000, -68.92770, -84.21145, 0.00000, 13.41981, -1.80156, 0.00000, 10.70956, -8.96349, 0.00000, 8.73474, -2.45083, 0.00000, 4.85948, -6.86615, 0.00000, 2.17993, -6.12375, 0.00000, 9.71606, -11.09536, 0.00000, 12.03736, -9.41146]
\ No newline at end of file
---
> value: [0.00000, -36.61945, -62.87399, 0.00000, -68.92770, -84.21145, 0.00000, 13.41981, -1.80156, 0.00000, 10.70956, -8.96349, 0.00000, 8.73474, -2.45083, 0.00000, 4.85948, -6.86615, 0.00000, 2.17993, -6.12375, 0.00000, 9.71606, -11.09536, 0.00000, 12.03736, -9.41146]
```
FIrst and 4th values are different: for both the code gives 2.0000, while the reference output file contains 0.0000. Can you double check this?

**Comment by jamatics** [comment]
Hi Anthony, 

I am not encountering this problem. I am attaching the screen below
Screenshot 2020-07-07 at 09 04 31
I am also getting fprintf error while compiling the code. 
Hope we can solve this problem. 

Thanks, 
Jawad

**Comment by aremazeilles** [comment]
The output you show in your screenshot (I assume `totalResult` is what would be written as ramgeofmotion pi score) contains 2.0 as first and 4th value, as I do. **But** the output score [stored in the repo](https://github.com/jamatics/pi_ctag/blob/run_pi_x/rangeofMotion/tests/data/output/Local_Score.yaml#L2) contains 0.0 as first and 4th value. 

2 options:

* the stored output in tests/data/output is correct, and code is wrong and has to be revised.
* the stored output in tests/data/output is wrong, and this file has to be revised.

Could you thus confirm whether the output pi vector should have v[0] = 0 or 2.0, and v[3]= 0 or 2.0?


Regarding your printf error, could it be that you launch the command with a folder name that does not exist? The program assumes the folder to write to results already exists.

**Comment by jamatics** [comment]
Hi Anthony, 

I am working on it. It seems there might be a bug in the code. Let me check in more detail. 

Thanks, 
Jawad

**Comment by jamatics** [comment]
Hi Anthony, 

I have checked and find two bugs: 
1. The upper and lower limit for the first two joints were not defined well as the provided data was measuring the values in negative. I would like to change these values in computePI.m for rangeofmotion pi. In short joint limits for left and right forearm will be changed to [-140, -10] and [-140, -10] from [10, 140] and [10, 140] respectively. 
2. I have also observed some corrections in romCheck.m file that is basically a function used by the main pi. 
With these two modifications, we can generate the same [output score stored in the repo](https://github.com/jamatics/pi_ctag/blob/run_pi_x/rangeofMotion/tests/data/output/Local_Score.yaml#L2) 
I can modify the code today and push it to branch run_pi_x or you prefer to push such changes to master? 

Thanks, 
Jawad

**Comment by aremazeilles** [comment]
Great!

I think it is fine to make the change in this branch (run_pi_x), and then, merge all.

**Comment by jamatics** [comment]
I have pushed changes to run_pi_x. Please verify if the PI is generating the correct results now. 
Thanks. 
Looking forward to contributing to EB 
Jawad

**Comment by aremazeilles** [comment]
Now I think this is ready for merge. Doing so right now!

**Pull request merged**